Mathlib/LinearAlgebra/Multilinear/Basic.lean

```diff
@@ -760,8 +763,151 @@ theorem domDomCongr_eq_iff (σ : ι₁ ≃ ι₂) (f g : MultilinearMap R (fun _
 
 end
 
+
+section Derivative
+
+/-! The goal of this part is, for `f` a multilinear map in finitely many variables indexed by `ι`,
+to develop `f(x + y)` as a finite formal multilinear series in `y`. The term of degree `n` of
+this formal multilinear series will be a sum over finsets `s` of `ι` of cardinality `n` of a
+multilinear map indexed by `s`, evaluated at `fun (_ : s) => y`; we use a linear order on
+`ι` to identify all such `s` to `Fin n`. We also give a more direct definition of the term
+of degree `1` in `f.linearDeriv`, which doesn't require a linear order on `ι`, and prove the
+equivalence of the two definitions.-/
+
+lemma domDomRestrict_aux [DecidableEq ι] (s : Set ι) [(i : ι) → Decidable (i ∈ s)]
+    [DecidableEq s] (x : (i : s.compl)→ M₁ i) (z : (i : s) → M₁ i) (i : s)
```

Comment:
Can you use the `compl` notation here, as in my original suggestion?

Reply:
If you mean the notation `sᶜ`, no I can't. I get an error message 
```failed to synthesize instance
  HasCompl (Type uι)
```
and I don't know how to get rid of it.

Reply:
```suggestion
    [DecidableEq s] (x : (i : ↥sᶜ)→ M₁ i) (z : (i : s) → M₁ i) (i : s)
```
should work, which tells lean you are casting from something that isn't a type.

Reply:
Yup it works, I changed it ! Thanks !